### PR TITLE
Introduce self-monitor dockerfile

### PR DESCRIPTION
**Pull request by rakesh-garimella**
**Description**

Changes proposed in this pull request:

- Create telemetry selfmonitor image via docker file

**Related issue(s)**

https://github.com/kyma-project/telemetry-manager/issues/871

### CODE REVIEW

File: self-monitor/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+FROM node:alpine3.18 as builder
```
Comment (a-thaler) [inline self-monitor/Dockerfile:1]: The newest alpine image is 3.19.1 ...

File: self-monitor/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+RUN apk add --no-cache git make musl-dev go curl bash
+
+# Configure Go
+ENV GOROOT /usr/lib/go
+ENV GOPATH /go
+ENV PATH /go/bin:$PATH
+
+RUN mkdir -p ${GOPATH}/src ${GOPATH}/bin
+WORKDIR /
+RUN curl -LO https://github.com/prometheus/prometheus/archive/refs/tags/v2.50.1.tar.gz  && \
```
Comment (a-thaler) [inline self-monitor/Dockerfile:11]: Please use the LTS versions: https://prometheus.io/docs/introduction/release-cycle/

**Comment by a-thaler** [comment]
CODEWONER files needs adjustement. Also think about prefixing the project with "telemetry-" as otherwise the context is not easy to get. Or make it a generic thing by calling it prometheus.

A README will help as well

File: self-monitor/Dockerfile (hunk 2 above)
Comment (rakesh-garimella) [reply self-monitor/Dockerfile:11]: done

File: telemetry-self-monitor/env-vars [hunk 3]
```diff
@@ -0,0 +1,2 @@
+PROMETHEUS_VERSION=2.45.4
```
Comment (a-thaler) [inline telemetry-self-monitor/env-vars:1]: the file is called envs for the other projects, I think the naming is relevant for the prow job

File: telemetry-self-monitor/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+WORKDIR /
+RUN curl -LO https://github.com/prometheus/prometheus/archive/refs/tags/v${PROMETHEUS_VERSION}.tar.gz  && \
+    tar xvzf v${PROMETHEUS_VERSION}.tar.gz
+
+WORKDIR /prometheus-${PROMETHEUS_VERSION}
+COPY plugins.yml plugins.yml
+RUN make build
+
+FROM alpine:${ALPINE_VERSION}
+
```
Comment (a-thaler) [inline telemetry-self-monitor/Dockerfile:22]: I think these command is important to install latest patches for the alpine libs
```suggestion
RUN apk --no-cache upgrade
```

File: telemetry-self-monitor/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Telemetry Self Monitor Docker Image
+
+The container image is built based on latest [prometheus lts image](https://prometheus.io/docs/introduction/release-cycle/).
```
Comment (NHingerl) [inline telemetry-self-monitor/README.md:3]:
```suggestion
The container image is built based on the latest [Prometheus LTS image](https://prometheus.io/docs/introduction/release-cycle/).
```

File: telemetry-self-monitor/README.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+Additionally, there is [plugins.yml](./plugins.yml) which contains the list of the plugins required.
```
Comment (NHingerl) [inline telemetry-self-monitor/README.md:5]:
```suggestion
Additionally, there is a [plugins.yml](./plugins.yml) file, which contains the list of the required plugins. 
```

File: telemetry-self-monitor/README.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Telemetry Self Monitor Docker Image
+
+The container image is built based on latest [prometheus lts image](https://prometheus.io/docs/introduction/release-cycle/).
+
+Additionally, there is [plugins.yml](./plugins.yml) which contains the list of the plugins required. 
+
+## Build locally
+
+To build the image locally, execute the following command, entering the proper versions taken from the `envs` file:
```
Comment (NHingerl) [inline telemetry-self-monitor/README.md:9]:
```suggestion
To build the image locally with the versions taken from the `envs` file, run:
```

File: telemetry-self-monitor/Dockerfile (hunk 4 above)
Comment (rakesh-garimella) [reply telemetry-self-monitor/Dockerfile:26]: done

**Review by a-thaler — DISMISSED** [review]

**Review by a-thaler — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**